art/estimators/classification/pytorch.py

```diff
@@ -484,13 +484,22 @@ def fit_generator(self, generator: "DataGenerator", nb_epochs: int = 20, **kwarg
 
     def clone_for_refitting(self) -> "PyTorchClassifier":  # lgtm [py/inheritance/incorrect-overridden-signature]
         """
-        Create a copy of the classifier that can be refit from scratch. Will inherit same architecture, optimizer and
-        initialization as cloned model, but without weights.
+        Create a copy of the classifier that can be refit from scratch. Will inherit same architecture, same type of
+        optimizer and initialization as the original classifier, but without weights.
 
         :return: new estimator
         """
         model = copy.deepcopy(self.model)
-        clone = type(self)(model, self._loss, self.input_shape, self.nb_classes, optimizer=self._optimizer)
+
+        if self._optimizer is None:  # pragma: no cover
+            raise ValueError("An optimizer is needed to train the model, but none for provided.")
+
+        # create a new optimizer that binds to the cloned model's parameters
+        new_optimizer = type(self._optimizer)(model.parameters(), lr=self._optimizer.defaults["lr"])  # type: ignore
```

Comment:
Check what lr is from `self._optimizer.defaults["lr"]`. We are expecting the actual `lr` used here.

Reply:
Yes, defaults contained the parameter passed in.
```
>>> import torch
>>> from torch import nn
>>> model = nn.Linear(100, 10)
>>> opt = torch.optim.Adam(model.parameters(), lr=0.1)
>>> opt.defaults
{'lr': 0.1, 'betas': (0.9, 0.999), 'eps': 1e-08, 'weight_decay': 0, 'amsgrad': False}
>>>
>>> opt2 = torch.optim.Adam(model.parameters())
>>> opt2.defaults
{'lr': 0.001, 'betas': (0.9, 0.999), 'eps': 1e-08, 'weight_decay': 0, 'amsgrad': False}
```